Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/Index.rst

```diff
@@ -0,0 +1,624 @@
+
+FloatValidator
+==============
+
+Checks if a value is a float.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Float'])]
+    protected float $price;
+
+..  _extbase-validator-image-dimensions:
+
+ImageDimensionsValidator
+========================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\ImageDimensionsValidator`
+validate image dimensions of a given :php-short:`\TYPO3\CMS\Core\Http\UploadedFile`
+:php-short:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` of `UploadedFile`
+objects.
+
+Options:
+
+`width`
+    The exact width of the image
+`height`
+    The exact height of the image
+`minWidth`
+    The minimum width of the image
+`maxWidth`
+    The maximum width of the image
+`minHeight`
+    The minimum height of the image
+`maxHeight`
+    The maximum height of the image
+
+..  include:: _NoExamples.rst.txt
+
+..  _extbase-validator-integer:
+
+IntegerValidator
+================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\IntegerValidator`
+ensures that a value is an integer.
+
+This validator is useful for validating numeric fields that must contain whole
+numbers, such as quantities, IDs, or counters.
+
+#[Validate(['validator' => 'Integer'])]
+protected mixed $quantity;
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Integer'])]
+    protected mixed $quantity;
+
+..  _extbase-validator-mime-type:
+
+MimeTypeValidator
+=================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\MimeTypeValidator`
+validates mime types of a given :php-short:`\TYPO3\CMS\Core\Http\UploadedFile`
+:php-short:`\TYPO3\CMS\Extbase\Persistence\ObjectStorage` of `UploadedFile`
+objects.
+
+Does also validate, if the given file extension matches allowed file extensions
+for the detected mime type.
+
+Options:
+
+`allowedMimeTypes`
+    Allowed mime types (using */* IANA media types)
+
+`ignoreFileExtensionCheck`
+    If set to "true", the file extension check is disabled.
+    Be aware of security considerations when setting this to "true".
+
+..  include:: _NoExamples.rst.txt
+
+..  _extbase-validator-notempty:
+
+NotEmptyValidator
+=================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\NotEmptyValidator`
+ensures that a value is not considered empty.
+
+"Empty" in this context means:
+
+-   An empty string (`''`)
+-   `null`
+-   An empty array (`[]`)
+
+This validator is commonly used to enforce required fields.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'NotEmpty'])]
+    protected string $title;
+
+..  _extbase-validator-numberrange:
+
+NumberRangeValidator
+====================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\NumberRangeValidator`
+checks that a number falls within a specified numeric range.
+
+This validator supports integers and floats and is useful for validating
+percentages, prices, limits, or any numeric input with minimum and/or maximum
+constraints.
+
+..  _extbase-validator-numberrange-options:
+
+Validator options
```

Comment:
The formatting for "Options" is different for some validators …

Reply:
Unified it